### PR TITLE
feat(service-portal): Generic License UI vol2

### PR DESCRIPTION
**Pull request by disaerna**
# Generic License UI for Service portal 💳 

Co-written with @thorkellmani 

* Rewoke license UI for service portal. A feature that was a part of earlier deploy but was removed last minute. 
* Logging changes for API's
* Mapping changes for licenses
* Add correct config values and secrets used
* ... + more 🤯 

A mix of old and new branches merged together to create this madness, hence the commits 🙈 

## What

Display all licenses available using generic license logic. 

* Vinnuvélaréttindi
* ADR réttindi
* Skotvopnaleyfi
* Ökuréttindi

## Why

More licenses 👍 
More future proof, minimal amount of effort when new licenses come in 👍 
Same or similar UI for all licenses 👍 

## Screenshots / Gifs
Overview: 
Screenshot 2022-09-29 at 10 28 08

Firearm license: 
Screenshot 2022-09-29 at 10 29 02
Firearm properties:
Screenshot 2022-09-29 at 10 29 18

Machine license: 
Screenshot 2022-09-29 at 10 28 29

ADR license: 

Screenshot 2022-09-29 at 10 28 18

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/service-portal/licenses/src/lib/messages.ts [hunk 1]
```diff
@@ -78,7 +126,7 @@ export const m = defineMessages({
   },
   errorFetchingDrivingLicense: {
```
Comment (thordurhhh) [inline libs/service-portal/licenses/src/lib/messages.ts:127]: Watch out for the contentful translation 👀

File: libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseService.api.ts [hunk 2]
```diff
@@ -22,73 +33,140 @@ export class GenericAdrLicenseApi implements GenericLicenseClient<AdrDto> {
...
+
+      return {
+        valid: false,
+        data: undefined,
+        error,
+      }
+    }
+
+    /*
+      TODO: VERIFICATION!!!!!!!! Máni (thorkellmani @ github)
```
Comment (thordurhhh) [inline libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseService.api.ts:164]: There are so many todo's @thorkellmani 

I wonder if they require a better explanation on why its a TODO and not being resolved in this PR. 🙏

**Review by thordurhhh — CHANGES_REQUESTED** [review]
Some minor issues that need to be taken care of 🌞  💻

File: libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseMapper.ts [hunk 3]
```diff
@@ -81,21 +81,27 @@ export const parseAdrLicensePayload = (
...
   )
 
   if (tankar) data.push(tankar)
 
-  const notTankar = parseRights(
-    '10. Annað en í tanki ',
-    adrRights.filter((field) => !field.tankar),
+  const grunn = parseRights(
+    label ? label['otherThanTanks'] : i18n.otherThanTanks[locale],
+    adrRights.filter((field) => field.grunn),
```
[the remark this answers is not in the document]
Comment (thorkellmani) [reply libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseMapper.ts:92]: oops lol

File: libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseService.api.ts (hunk 2 above)
Comment (thorkellmani) [reply libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseService.api.ts:164]: improved the todos

File: libs/service-portal/licenses/src/components/QRCodeModal/PkPass.tsx [hunk 4]
```diff
@@ -131,7 +137,7 @@ export const PkPass = ({ expireDate, textButton = false }: PkPassProps) => {
         </>
       )}
 
-      {isMobile && (
+      {(isMobile || !isDriversLicense) && (
```
[the remark this answers is not in the document]
Comment (disaerna) [reply libs/service-portal/licenses/src/components/QRCodeModal/PkPass.tsx:140]: Yes 💯

File: libs/service-portal/licenses/src/screens/LicensesOverview/LicensesOverview.tsx [hunk 5]
```diff
@@ -1,26 +1,141 @@
...
+   * Please remove all code when fully released.
+   */
+  const featureFlagClient: FeatureFlagClient = useFeatureFlagClient()
+  const [licenseTypes, setLicenseTypes] = useState<Array<GenericLicenseType>>([
+    GenericLicenseType.DriversLicense,
+    GenericLicenseType.AdrLicense,
+    GenericLicenseType.MachineLicense,
+    GenericLicenseType.FirearmLicense,
+  ])
+  // useEffect(() => {
```
[the remark this answers is not in the document]
Comment (disaerna) [reply libs/service-portal/licenses/src/screens/LicensesOverview/LicensesOverview.tsx:106]: Ohno 🙈

**Review by peturgq — APPROVED** [review]

**Review by thordurhhh — APPROVED** [review]
💯

File: apps/api/src/app/app.module.ts [hunk 6]
```diff
@@ -220,8 +220,17 @@ const autoSchemaFile = environment.production
...
+      },
+      machineLicense: {
+        apiKey: environment.machineLicense.pkPassApiKey!,
+        apiUrl: environment.smartSolutionsApiUrl!,
+        passTemplateId: environment.machineLicense.passTemplateId!,
+      },
+      adrLicense: {
+        apiKey: environment.adrLicense.pkPassApiKey!,
+        apiUrl: environment.smartSolutionsApiUrl!,
+        passTemplateId: environment.adrLicense.passTemplateId!,
```
Comment (eirikurn) [inline apps/api/src/app/app.module.ts:233]: Please put on the schedule to change this module (and others you maintain) from the `Module.register(environment),` pattern to just `Module,` using ConfigModule instead.

We'd like to get rid of environment files completely.

**Review by eirikurn — APPROVED** [review]
Won't block this from the release, but would appreciate a config cleanup in followup. 👍

File: apps/api/src/app/app.module.ts (hunk 6 above)
Comment (thorkellmani) [reply apps/api/src/app/app.module.ts:233]: a refactor is next up

**Pull request merged**